inspect/inspect.go

```diff
@@ -0,0 +1,174 @@
+package inspect
+
+import (
+	"context"
+	"errors"
+	"net"
+	"sync"
+
+	cfg "github.com/tendermint/tendermint/config"
+	inspect_rpc "github.com/tendermint/tendermint/inspect/rpc"
+	"github.com/tendermint/tendermint/libs/log"
+	tmstrings "github.com/tendermint/tendermint/libs/strings"
+	rpccore "github.com/tendermint/tendermint/rpc/core"
+	sm "github.com/tendermint/tendermint/state"
+	"github.com/tendermint/tendermint/state/indexer"
+	"github.com/tendermint/tendermint/state/indexer/sink"
+	"github.com/tendermint/tendermint/store"
+	"github.com/tendermint/tendermint/types"
+)
+
+// Inspect manages an RPC service that exports methods to debug a failed node.
+// After a node shuts down due to a consensus failure, it will no longer start
+// up its state cannot easily be inspected. An Inspect value provides a similar interface
+// to the node, using the underlying Tendermint data stores, without bringing up
+// any other components. A caller can query the Inspect service to inspect the
+// persisted state and debug the failure.
+type Inspect struct {
+	routes rpccore.RoutesMap
+
+	rpcConfig *cfg.RPCConfig
+
+	indexerService *indexer.Service
+	eventBus       *types.EventBus
+	logger         log.Logger
+}
+
+// New constructs a new Inspect from the passed in parameters.
```

Comment:
I feel like these kinds of docs strings exist to satisfy linters, and should really be omitted, or expanded to be useful.

Reply:
I agree, but I think the correct fix is to make the comment more explanatory. There's almost always some useful information that can be included to make the doc more useful than restating the name.

(If we are avoiding things just to silence the linter, we should arguably fix the linter settings)

Perhaps in this case:

```go
// New constructs a new unstarted inspector that serves state from the specified
// block and state stores. The inspector does not modify the stores it inspects.
// The event sinks are used to (whatever they're used to do).
// The caller is responsible for starting and stopping the inspector service.
```

Or whatever may be true. Right now, I agree, the comment doesn't say much of use.

Reply:
done.